### PR TITLE
Make packaging tests use jdk downloader

### PR DESCRIPTION
**Pull request by rjernst**
This commit removes the jdk11 download in vagrant provisioning and
converts it to using the jdk downloader for the system jdk, and sets up
a separate jdk for use by the test (which will be converted to running
gradle in a followup).

### CODE REVIEW

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantTestPlugin.groovy [hunk 1]
```diff
@@ -280,19 +314,23 @@ class VagrantTestPlugin implements Plugin<Project> {
...
                        test_args=( "${-> project.extensions.esvagrant.testClass}" )
                      else
                        test_args=( "\$@" )
                      fi
-                     "\$SYSTEM_JAVA_HOME"/bin/java -cp "\$PACKAGING_TESTS/*" org.elasticsearch.packaging.VMTestRunner "\${test_args[@]}"
+                     
+                     if [ -z "\$SYSTEM_JAVA_HOME" ]; then
+                       export SYSTEM_JAVA_HOME="${-> convertPath(project, linuxSystemJdk.toString()) }"
+                     fi
+                     "${-> convertPath(project, linuxGradleJdk.toString()) }"/bin/java -cp "\$PACKAGING_TESTS/*" org.elasticsearch.packaging.VMTestRunner "\${test_args[@]}"
```
Comment (mark-vieira) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantTestPlugin.groovy]: As I understand it we are on the path to ditch this `VMTestRunner` in favor of just delegating to Gradle, yes?

**Review by mark-vieira — APPROVED** [review]
LGTM

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantTestPlugin.groovy (hunk 1 above)
Comment (rjernst) [reply buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantTestPlugin.groovy]: Yep, just trying to do this in phases.

**Comment by rjernst** [comment]
@elasticmachine run elasticsearch-ci/1

**Pull request merged**